src/Templates/ModelTemplate.cshtml

```diff
@@ -344,7 +344,7 @@ else
 @if (Model is PageTypeGo modelPageType)
 {
     var itemName = modelPageType.ItemName;
-    var itemType = modelPageType.ElementType.Name;
+    var itemType = modelPageType.ElementType.HasInterface() ? modelPageType.ElementType.GetInterfaceName() : modelPageType.ElementType.Name.Value;
```

Comment:
I think this should be added to the PageTypeGo class as it basically duplicates line 32 in the ctor.

Reply:
thought it might be confusing to have both Name and type as properties in the Page class, but I named it ReturnTypeName, should be fine.